LevelComponents/EntitySet.cpp

```diff
@@ -59,11 +78,173 @@ namespace LevelComponents
     /// <summary>
     /// Get a copy of EntitySetinfoTableElement from this EntitySet.
     /// </summary>
-    std::vector<EntitySetinfoTableElement> EntitySet::GetEntityTable()
+    QVector<EntitySetinfoTableElement> EntitySet::GetEntityTable() const
     {
-        std::vector<EntitySetinfoTableElement> newtable;
-        newtable.assign(EntityinfoTable.begin(), EntityinfoTable.end());
+        QVector<EntitySetinfoTableElement> newtable(EntityinfoTable);
         return newtable;
     }
 
+    /// <summary>
+    /// Render the whole Entityset using one palette.
+    /// </summary>
+    /// <param name="palNum">
+    /// Palette number used to render the current entityset.
+    /// </param>
+    QPixmap EntitySet::GetPixmap(const int palNum)
+    {
+        // Initialize the palettes
+        ResetPalettes();
```

Comment:
Why do you need to reset both of these just to render the pixmap?

Reply:
so i don't need to always call some extra function when appply some changes to palettes or tiles and GetPixmap

Reply:
I think that these should only be called wherever you make changes to palettes and tiles, not as part of the render function

Reply:
As discussed, this is okay because they are helper functions for rendering